pages/_app.js

```diff
@@ -19,14 +19,31 @@ export default function App({ Component, pageProps }) {
       ...oldTasks,
     ]);
   }
+  function updateTask(taskId, updatedTask) {
+    setTasks((oldTasks) => {
+      const newTasks = oldTasks.map((oldTask) => {
+        if (oldTask.id === taskId) {
+          return updatedTask;
+        } else {
+          return oldTask;
+        }
+      });
+      return newTasks;
```

Comment:
Well, it's an extra step to save it to a const before returning it.
But works fine and it's readable.

Reply:
Exactly, readability was important for me here.